pkg/oauthserver/oauth/external/gitlab/gitlab_oidc.go

```diff
@@ -0,0 +1,97 @@
+package gitlab
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+	"regexp"
+	"strconv"
+
+	"github.com/openshift/origin/pkg/oauthserver/oauth/external"
+	"github.com/openshift/origin/pkg/oauthserver/oauth/external/openid"
+)
+
+const (
+	// https://gitlab.com/help/integration/openid_connect_provider.md
+	// Uses GitLab OIDC, requires GitLab 11.1.0 or higher
+	// Earlier versions do not work: https://gitlab.com/gitlab-org/gitlab-ce/issues/47791#note_81269161
+	gitlabAuthorizePath = "/oauth/authorize"
```

Comment:
is there a good reason to prepend all these module variables with "gitlab" ?

Reply:
Matches the existing code and the other provider integrations.